pulpcore/cli/ansible/distribution.py

```diff
@@ -112,16 +114,18 @@ def update(
             repo = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
             if version is not None:
                 if dist_body["repository"]:
-                    distribution_ctx.update(href, body={"repository": ""})
+                    pulp_ctx.wait_for_task(distribution_ctx.update(href, body={"repository": ""}))
```

Comment:
This is because distributions cannot be updated "in one go" (and both file and rpm probably don't even have this problemas they facilitate publications)? I consider this a bug in pulpcore. And workarounds for bugs in pulpcore belong into the corresponding context classes.
Using `pulp_ctx.wait_for_task` is mixing several levels of abstraction together.

And the much bigger problem is, this is lying to the user, as it is not performing stuff in the background as requested.
i.e. if the resource is blocked by another long running task, the command will wait for that task, and then issue the second, which will very likely be executed immediately.
Does the pulp tasking system not allow to queue up both tasks, and they will be executed in that order yielding the expected result without awaiting any of them?
If all of this does not work as expected, i'd rather issue an error, that this command is not capable of "running in the background".

Reply:
According to dalley and david the tasks will be queued up and executed in order if they operate on the same resource. So I don't need the wait_for_task call after all, I'll get rid of them.